### DISCUSSION TITLE
Retrodeck collaboration

### DISCUSSION THREAD

**Discussion by zurdi15** [rommapp/romm#252] [Ideas]
Hi @XargonWan,

As I [said](https://github.com/zurdi15/romm/issues/230#issuecomment-1551770775), we should find a place to talk about the most common design problems for this ecosystem.

I pick up the proposal you made me in the past.

**Comment by Typhonragewind** [comment]
> Well first of all, sorry for my bad english as it is not my main language:
> 
> We talked in the past about [collaborating togheter](https://github.com/zurdi15/romm/issues/30) to make a stronger software ecosystem but I refused because the nature of my personal situation makes me having some periods of hard work and some periods of not having time to put in any kind of personal project, being unable to collaborate or having a compromise with other developers. This was also empowered by the fact that the code base of RomM still needs a lot of work in terms of design and refactor.
> 
> However this projects is getting so much bigger than I thought at first, so I have been wondering for several days about having a collaboration with @XargonWan at least in terms of designing the same solutions for the most common problems (savefiles, folder structures, dlc/updates management, etc) and setting some standards in the emulation/games preservation community. He helped me a lot and he already have been analyzing some of the most common problems/features the users are asking for.
> 
> I think the wind is changing :)
> 
> I am opening a [discussion](https://github.com/zurdi15/romm/discussions/252) to not hijack this issue

Your English is awesome, don't worry about it!

I'm glad to see this project grow so much, i find it really amazing. And i'm happy that you're reconsidering an alliance withe RetroDeck :D

**Comment by XargonWan** [comment]
Actually a RetroDECK integration may not be needed as RomM is managing roms and RetroDECK is playing them: you can use RomM from your Steam Deck with no issues already. 
However if a nice integration may come handy in the future I'm not denying it. 

My collaboration proposal was actually to work on a new project named [RetroMan (Retro Manager)](https://github.com/RetroMan-project), but unfortunately the other developer left the project and my skills in programming are not so advanced to allow me to do this project on my own. Let's say that I am more a Project Manager than a programmer. 

So RetroMan may be dead even before starting. 
But we got RomM, and this is great, and my question is now: @zurdi15, are you interested into expanding the scope of RomM with the RetroMan signature features or you want to follow your scope?

In any cases I will help however I can :)

**Reply by gantoine** [reply]
I took a look over the planned features, I'd love to see some of these features make their way into RomM:

- Multi device support
- Synchronize ROMs/saves to devices (local copy, s/ftp, smb)
- Saves backups (over the air)
- Duplicates management
- CLI support

A cursory google search returns a number of abandoned ROM manager projects; I'd like to avoid that kind of fragmentation, and get more hands working on a single project (ideally this one).

Furthermore if we can align on the direction this project takes, and the scope it should cover, I'm happy to shoulder the burden of the engineering work needed to get us there! Once that's done, we can discuss some engineering changes that should happen to allow for things like a CLI, REST endpoints, native clients, etc...

**Reply by XargonWan** [reply]
Feel free to pick whatever you like from RetroMan an maniuplate it, then we can dicuss it. 
I don't like fragmentation as well so we got another common point here. 

EDIT: feel free to ask me about the obscure points so I can explain better what was my idea. We can even do a meeting one day.

**Reply by XargonWan** [reply]
I started to open some discussions so we can tweak the ideas and then create some issues.

**Reply by gantoine** [reply]
@XargonWan I sent you (or someone with your name) a friend request on Discord, let's find some time to chat more deeply about your ideas soon.

**Comment by gantoine** [comment]
@XargonWan and I chat a great chat over discord earlier today where he went over the idea behind, and some of the proposed features of, [Retroman](https://github.com/RetroMan-project). I'm currently working on a "pitch" for this project: a high-level overview of its main purpose, a review of the code base, and future functionally we can slowly build towards.

Given all of that, I'm going to close this discussion, and we can start to work on a collaborative document detailing this project.
